lib/Sema/TypeCheckAccess.cpp

```diff
@@ -803,29 +813,31 @@ class AccessControlChecker : public AccessControlCheckerBase,
           complainRepr = thisComplainRepr;
           protocolControlErrorKind = PCEK_Refine;
           downgradeToWarning = downgradeDiag;
+          declKind = declKindForType(requirement.getType());
         }
       });
     });
 
-    checkRequirementAccess(proto,
-                           proto->getFormalAccessScope(),
-                           proto->getDeclContext(),
-                           [&](AccessScope typeAccessScope,
-                               const TypeRepr *thisComplainRepr,
-                               DowngradeToWarning downgradeDiag) {
-      if (typeAccessScope.isChildOf(minAccessScope) ||
-          (!complainRepr &&
-           typeAccessScope.hasEqualDeclContextWith(minAccessScope))) {
-        minAccessScope = typeAccessScope;
-        complainRepr = thisComplainRepr;
-        protocolControlErrorKind = PCEK_Requirement;
-        downgradeToWarning = downgradeDiag;
-
-        // Swift versions before 5.0 did not check requirements on the
-        // protocol's where clause, so emit a warning.
-        if (!TC.Context.isSwiftVersionAtLeast(5))
-          downgradeToWarning = DowngradeToWarning::Yes;
-      }
+    forAllRequirementTypes(proto, [&](Type type, TypeRepr *typeRepr) {
```

Comment:
Can you clarify why this change? As far as I can tell the two invocations are equivalent, with this one spelling more things out explicitly.

Reply:
The `forAllRequirementTypes` callback doesn't include the actual type, and its other clients don't need it. This just makes it a bit easier to unify the `declKindForType` logic.

Reply:
Oh, I see. And this will do the right thing for simple types and something sensible for other types. Okay, thanks for explaining!